src/frameworks/index.ts

```diff
@@ -411,7 +411,12 @@ You can link a Web app to a Hosting site here https://console.firebase.google.co
         },
       ]);
 
-      if (!targetNames.includes("functions")) targetNames.unshift("functions");
+      if (!targetNames.includes("functions")) {
+        targetNames.unshift("functions");
+        if (options.only) {
```

Comment:
May want to make sure `functions` isn't included already in `only`. I'm not sure what the ramifications would be off hand.